warehouse/internal/repo/repo.go

```diff
@@ -18,3 +19,21 @@ func WithNow(now func() time.Time) Opt {
 		r.now = now
 	}
 }
+
+func (r *repo) WithTx(f func(tx *sqlmiddleware.Tx) error) error {
+	tx, err := r.db.Begin()
+	if err != nil {
+		return fmt.Errorf("begin transaction: %w", err)
+	}
+
+	if err := f(tx); err != nil {
+		if rollbackErr := tx.Rollback(); rollbackErr != nil {
+			return fmt.Errorf("rollback transaction for %w: %w", err, rollbackErr)
```

Comment:
This is like Nando's double :chicken: wrap but with errors :sweat_smile:

Reply:
I am a bit skeptical here. Should we ignore rollback error?

Reply:
Nah, it's OK. It works perfectly fine and it is possible to unwrap both errors :+1:

Reply:
I wasn't aware that is possible to wrap two errors that the same time, cool